### PR TITLE
Update Xcode and configure.ac files

### PR DESCRIPTION
**Pull request by AtariDreams**
Regenerated Xcode by updating its settings, and autoupdated configure.ac

### CODE REVIEW

**Comment by michaelrsweet** [comment]
@AtariDreams CUPS' configure script CANNOT be regenerated properly using aclocal. That said, this PR only contains changes to the generated configure script and not to the source - I'm not sure what changed?

**Comment by AtariDreams** [comment]
@michaelrsweet https://github.com/OpenPrinting/cups/commit/63e2ec255304e78fa5fc245165d67b08b35fa806 this commit was made back in May, but the configure wasn't regenerated then, hence why it changed now

File: config-scripts/cups-compiler.m4 [hunk 1]
```diff
@@ -114,8 +114,8 @@ AS_IF([test -n "$GCC"], [
 
     # Default optimization options...
     AS_IF([test -z "$OPTIM"], [
-	# Default to optimize-for-size and debug
-	OPTIM="-Os -g"
+	# Default to optimize-for-size and LTO
```
Comment (michaelrsweet) [inline config-scripts/cups-compiler.m4:117]: LTO isn't generally available, so I'd rather keep with the current default and let distros/packagers tweak things for their particular versions of GCC/Clang/etc.

File: xcode/CUPS.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -11582,6 +11592,7 @@
 					.,
 					..,
 				);
+				LLVM_LTO = YES;
```
Comment (michaelrsweet) [inline xcode/CUPS.xcodeproj/project.pbxproj:11595]: This doesn't hurt but doesn't help - the Xcode project is more to make it easier to do Clang static analysis of the code than to build it for macOS...

**Review by michaelrsweet — CHANGES_REQUESTED** [review]
See the comments - I don't want to add LTO to the configure options as not all compilers support it.

**Comment by AtariDreams** [comment]
Almost all compilers do actually. Only the really old ones don’t. That being said, I’ll gladly revert the change.

**Comment by AtariDreams** [comment]
All issues addressed!

File: config-scripts/cups-compiler.m4 [hunk 3]
```diff
@@ -114,8 +114,8 @@ AS_IF([test -n "$GCC"], [
 
     # Default optimization options...
     AS_IF([test -z "$OPTIM"], [
-	# Default to optimize-for-size and debug
-	OPTIM="-Os -g"
+	# Default to optimize-for-size
```
Comment (michaelrsweet) [inline config-scripts/cups-compiler.m4:117]: We still want debug symbols. Most Linux distributions will generate a debug package to help developers and then strip.

**Review by michaelrsweet — CHANGES_REQUESTED** [review]
We still want debug symbols so that crashes can show proper backtraces.

**Comment by AtariDreams** [comment]
Addressed!

File: Makefile [hunk 4]
```diff
@@ -42,7 +42,7 @@ all:
 	echo Using LIBS="$(LIBS)"
 	for dir in $(DIRS); do\
 		echo Making all in $$dir... ;\
-		(cd $$dir ; $(MAKE) $(MFLAGS) all $(UNITTESTS)) || exit 1;\
+		($(MAKE) -C $$dir $(MFLAGS) all $(UNITTESTS)) || exit 1;\
```
Comment (michaelrsweet) [inline Makefile:45]: "-C" is a GNU extension and this change will break builds on a LOT of systems. See the [POSIX make](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/make.html) documentation for a list of supported arguments.

File: Makefile [hunk 5]
```diff
@@ -61,7 +61,7 @@ libs:
 	echo Using LIBS="$(LIBS)"
 	for dir in $(DIRS); do\
 		echo Making libraries in $$dir... ;\
-		(cd $$dir ; $(MAKE) $(MFLAGS) libs) || exit 1;\
+		($(MAKE) -C $$dir $(MFLAGS) libs) || exit 1;\
```
Comment (michaelrsweet) [inline Makefile:64]: Ditto

File: Makefile [hunk 6]
```diff
@@ -80,7 +80,7 @@ unittests:
 	echo Using LIBS="$(LIBS)"
 	for dir in $(DIRS); do\
 		echo Making all in $$dir... ;\
-		(cd $$dir ; $(MAKE) $(MFLAGS) unittests) || exit 1;\
+		($(MAKE) -C $$dir $(MFLAGS) unittests) || exit 1;\
```
Comment (michaelrsweet) [inline Makefile:83]: Ditto

File: Makefile [hunk 7]
```diff
@@ -91,7 +91,7 @@ unittests:
 clean:
 	for dir in $(DIRS); do\
 		echo Cleaning in $$dir... ;\
-		(cd $$dir; $(MAKE) $(MFLAGS) clean) || exit 1;\
+		($(MAKE) -C $$dir $(MFLAGS) clean) || exit 1;\
```
Comment (michaelrsweet) [inline Makefile:94]: Ditto

File: Makefile [hunk 8]
```diff
@@ -121,7 +121,7 @@ distclean:	clean
 depend:
 	for dir in $(DIRS); do\
 		echo Making dependencies in $$dir... ;\
-		(cd $$dir; $(MAKE) $(MFLAGS) depend) || exit 1;\
+		($(MAKE) -C $$dir $(MFLAGS) depend) || exit 1;\
```
Comment (michaelrsweet) [inline Makefile:124]: Ditto

File: Makefile [hunk 9]
```diff
@@ -166,10 +166,10 @@ install:	install-data install-headers install-libs install-exec
 
 install-data:
 	echo Making all in cups...
-	(cd cups; $(MAKE) $(MFLAGS) all)
+	($(MAKE) -C cups $(MFLAGS) all)
```
Comment (michaelrsweet) [inline Makefile:169]: Ditto

File: Makefile [hunk 10]
```diff
@@ -166,10 +166,10 @@ install:	install-data install-headers install-libs install-exec
...
 	for dir in $(DIRS); do\
 		echo Installing data files in $$dir... ;\
-		(cd $$dir; $(MAKE) $(MFLAGS) install-data) || exit 1;\
+		($(MAKE) -C $$dir $(MFLAGS) install-data) || exit 1;\
```
Comment (michaelrsweet) [inline Makefile:172]: Ditto

File: Makefile [hunk 11]
```diff
@@ -186,7 +186,7 @@ install-data:
 install-headers:
 	for dir in $(DIRS); do\
 		echo Installing header files in $$dir... ;\
-		(cd $$dir; $(MAKE) $(MFLAGS) install-headers) || exit 1;\
+		($(MAKE) -C $$dir $(MFLAGS) install-headers) || exit 1;\
```
Comment (michaelrsweet) [inline Makefile:189]: Ditto

File: Makefile [hunk 12]
```diff
@@ -202,7 +202,7 @@ install-headers:
 install-exec:	all
 	for dir in $(DIRS); do\
 		echo Installing programs in $$dir... ;\
-		(cd $$dir; $(MAKE) $(MFLAGS) install-exec) || exit 1;\
+		($(MAKE) -C $$dir $(MFLAGS) install-exec) || exit 1;\
```
Comment (michaelrsweet) [inline Makefile:205]: Ditto

File: backend/Makefile [hunk 13]
```diff
@@ -65,7 +65,7 @@ all:	$(TARGETS)
 # Make library targets...
 #
 
-libs:
+libs: 		all
```
Comment (michaelrsweet) [inline backend/Makefile:68]: Why?

File: backend/Makefile [hunk 14]
```diff
@@ -162,7 +162,6 @@ install-headers:
 
 install-libs:
 
-
```
Comment (michaelrsweet) [inline backend/Makefile:165]: 2 blank lines please...

File: cgi-bin/Makefile [hunk 15]
```diff
@@ -58,7 +58,7 @@ all:	$(TARGETS)
 # Make library targets...
 #
 
-libs:
+libs:	$(LIBTARGETS)
```
Comment (michaelrsweet) [inline cgi-bin/Makefile:61]: Why? We no longer install libcupscgi as a shared library. The "libs" target is for installed libraries.

File: configure.ac [hunk 16]
```diff
@@ -9,11 +9,11 @@ dnl Licensed under Apache License v2.0.  See the file "LICENSE" for more
...
 dnl
 
-dnl We need at least autoconf 2.60...
-AC_PREREQ([2.60])
+dnl We need at least autoconf 2.71...
+AC_PREREQ([2.71])
 
 dnl Package name and version...
-AC_INIT([CUPS], [2.4.0], [https://github.com/openprinting/cups/issues], [cups], [https://openprinting.github.io/cups])
+AC_INIT([CUPS],[2.4.0],[https://github.com/openprinting/cups/issues],[cups],[https://openprinting.github.io/cups])
```
Comment (michaelrsweet) [inline configure.ac:16]: Any reason for removing the spaces?

File: ppdc/Makefile [hunk 17]
```diff
@@ -78,7 +78,7 @@ all:		$(TARGETS)
 # Make library targets...
 #
 
-libs:
+libs:		$(LIBTARGETS)
```
Comment (michaelrsweet) [inline ppdc/Makefile:81]: Same as for the cgi-bin makefile - there are no installed libraries here.

File: scheduler/Makefile [hunk 18]
```diff
@@ -91,7 +91,7 @@ all:		$(TARGETS)
 # Make library targets...
 #
 
-libs:
+libs:		$(LIBTARGETS)
```
Comment (michaelrsweet) [inline scheduler/Makefile:94]: No installed libraries.

**Review by michaelrsweet — CHANGES_REQUESTED** [review]
Some issues with the makefile changes - we need to keep the makefiles POSIX compliant, otherwise we break the build on a LOT of systems.

Also, the "libs" targets don't need to depend on the LIBTARGETS if they are all static libraries, since they are not installed.

**Comment by AtariDreams** [comment]
@michaelrsweet All issues resolved

File: configure.ac (hunk 16 above)
Comment (AtariDreams) [reply configure.ac:16]: autoupdate side effect

Comment (michaelrsweet) [reply configure.ac:16]: @AtariDreams Never use autoupdate on CUPS, it can break things... :/

**Review by michaelrsweet — APPROVED** [review]

**Pull request merged**